Traceback (most recent call last):
  File "/tmp/ipykernel_723269/4088739247.py", line 32, in train
    X_train = pd.get_dummies(X_train)
UnboundLocalError: local variable 'X_train' referenced before assignment